data/logs_model/logs_producer/splunk_logs_producer.py

```diff
@@ -22,11 +23,33 @@ def __init__(
         url_scheme="https",
         verify_ssl=True,
         index_prefix=None,
+        ssl_ca_path=None,
     ):
+        connect_args = {
+            "host": host,
+            "port": port,
+            "token": bearer_token,
+            "scheme": url_scheme,
+            "verify": verify_ssl,
+        }
+        if verify_ssl:
+            # Create an SSLContext object for cert validation
+            context = ssl.create_default_context()
+            context.check_hostname = True
+            context.verify_mode = ssl.CERT_REQUIRED
+            context.load_cert_chain(certfile=ssl_ca_path)
+            connect_args["context"] = context
+        else:
+            connect_args["verify"] = False
+
         try:
-            service = client.connect(
-                host=host, port=port, token=bearer_token, scheme=url_scheme, verify=verify_ssl
-            )
+            service = client.connect(**connect_args)
+        except TypeError as ex:
```

Comment:
when should we expect a `TypeError`? 🤔

Reply:
Interesting. In that case, maybe we should also treat the `FileNotFoundError` here?
I think this syntax should work but please double check:
```
except TypeError, FileNotFoundError as ex:
```

Also, a nitpick: in python we usually call an exception `e` (instead of `ex`), for example:
```
except TypeError as e:
```